resources/views/courses/action.blade.php

```diff
@@ -2,7 +2,7 @@
     @if (Auth::user()->student->isEnrolledInCourse($course))
         <button type="button" class="btn btn-outline-secondary btn-sm disabled">Enrolled</button>
 
-        @if ($settings->withinEnrollmentPeriod())
+        @if (app('settings')->withinExchangePeriod() && is_null(Auth::student()->getEnrollmentByCourse($course)->shift))
```

Comment:
`$settings`. This variable is injected by the ViewComposer.

You are putting important logic in views. I like the `isDeletable()` approach, but in the `Enrollment` model. The purpose of this method is to check the second condition.